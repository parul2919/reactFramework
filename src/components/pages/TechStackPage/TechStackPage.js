import React from "react";
import TechStack from "../../template/TechStack/TechStack";

const wrapperStyles = {
  maxWidth: "100%",
  margin: "auto",
};

const TechStackPage = () => (
  <div style={wrapperStyles}>
    <TechStack />
  </div>
);

export default TechStackPage;
