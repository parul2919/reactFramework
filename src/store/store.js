import { applyMiddleware, createStore } from "redux";
// Logger with default options
// import logger from 'redux-logger';
import createSagaMiddleware from "redux-saga";
import rootReducer from "./reducers/rootReducer";
import rootSaga from "./sagas/rootSaga";

const sagaMiddleware = createSagaMiddleware();
const store = createStore(
  rootReducer,
  applyMiddleware(sagaMiddleware)
  // applyMiddleware(sagaMiddleware, logger),
);
sagaMiddleware.run(rootSaga);

export default store;
