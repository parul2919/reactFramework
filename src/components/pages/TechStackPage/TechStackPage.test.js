import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../../store/store";
import TechStackPage from "./TechStackPage";

describe("<Component /> spec", () => {
  it("renders the component", () => {
    const view = render(
      <Provider store={store}>
        <Router>
          <TechStackPage />
        </Router>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});
