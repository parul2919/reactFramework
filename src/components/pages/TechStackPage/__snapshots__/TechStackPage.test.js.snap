// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="max-width: 100%; margin: auto;"
      >
        <div
          class="sc-hKMtZM lbYHHX dummy-template"
        >
          <header
            class="header sc-gsnTZi jOpTsK"
          >
            <nav
              class="navbar navbar-expand d-flex align-items-center justify-content-between"
            >
              <a
                class="navbar-brand "
                href="/"
              >
                <img
                  alt="logo-img"
                  src="test-file-stub"
                  width="150"
                />
              </a>
              <div
                class="collapse navbar-collapse"
              >
                <ul
                  class="navbar-nav ml-auto "
                >
                  <a
                    class="nav-link"
                    href="/"
                    style="text-decoration: underline;"
                    target="_self"
                  >
                    Tech Stacks
                  </a>
                  <a
                    class="nav-link"
                    href="/component-library"
                    style="text-decoration: none;"
                    target="_self"
                  >
                    Component Library
                  </a>
                  <a
                    class="nav-link"
                    href="/project-structure"
                    style="text-decoration: none;"
                    target="_self"
                  >
                    Project Structure
                  </a>
                </ul>
              </div>
            </nav>
          </header>
          <section
            class="wrapper-std"
          >
            <section
              class="techStack sc-dkzDqf ivVfSh"
            >
              <div
                class="techStack-list"
              >
                <h2>
                  Do find the tech-stack list which we are using to build this architecure
                </h2>
                <h3>
                  1. Interface Design Structure
                </h3>
                <p>
                  <a
                    href="https://atomicdesign.bradfrost.com/chapter-2/"
                  >
                    Atomic Design Methodology
                  </a>
                  - Atoms, molecules, organisms, templates, and pages
                </p>
                <p>
                  Atomic design allows us to see our UIs broken down to their atomic elements, and also allows us to simultaneously step through how those elements join together to form our final UIs and gives thoughtful, deliberate interface design systems.
                   
                </p>
                <h3>
                  2. Framwork Dependency
                </h3>
                <ul>
                  <li>
                    <a
                      href="https://reactjs.org/"
                    >
                      React
                    </a>
                     - Lightweight FE library that deals with the view layer.
                  </li>
                  <li>
                    <a
                      href="https://redux.js.org/"
                    >
                      Redux
                    </a>
                     - For State Management
                  </li>
                  <li>
                    <a
                      href="https://redux-saga.js.org/"
                    >
                      Redux Saga
                    </a>
                     - easier to manage, more efficient to execute, easy to test, and better at handling failures.
                  </li>
                  <li>
                    <a
                      href="https://reactrouter.com/"
                    >
                      React Router
                    </a>
                     - React Router is an API for React applications which provides dynamic, client-side routing, allows us to build a single-page web application with navigation without the page refreshing as the user navigates.
                  </li>
                  <li>
                    <a
                      href="https://styled-components.com/docs"
                    >
                      Styled components
                    </a>
                     - provides unique class names for your styles, thus eliminating the problems with class names duplications, misspellings, and overlaps.
                  </li>
                  <li>
                    <a
                      href="https://styled-components.com/docs/advanced"
                    >
                      ThemeProvider
                    </a>
                    - a wrapper component available in styled-components, we can quickly set up multiple custom themes in React and switch between them with ease.
                  </li>
                  <li>
                    <a
                      href="https://mui.com/material-ui/getting-started/overview/"
                    >
                      Material Ui
                    </a>
                    - CSS framework that provides React components
                  </li>
                  <li>
                    <a
                      href="https://react-hook-form.com/"
                    >
                      React Hook Form
                    </a>
                    - To build form components. Why React Hook Forms is because it isolates the component and avoids the other components from re-rending and thus helps improving performance also gives better validation support.
                  </li>
                </ul>
                <h3>
                  3. Quality Gates
                </h3>
                <ul>
                  <li>
                    <a
                      href="https://testing-library.com/docs/react-testing-library/setup/"
                    >
                      React Testing Liberary with jest
                    </a>
                    - React Testing Library is a testing utility tool that's built to test the actual DOM tree rendered by React on the browser. The goal of the library is to help you write tests that resemble how a user would use your application.
                  </li>
                  <li>
                    <a
                      href="https://eslint.org/"
                    >
                      Eslint 
                    </a>
                    :: .eslint.config.js - ESLint is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code, with the goal of making code more consistent and avoiding bugs.
                  </li>
                  <li>
                    <a
                      href="https://prettier.io/docs/en/"
                    >
                      Prettier
                    </a>
                    - Styling code conforms to a consistent style.
                  </li>
                  <li>
                    <a
                      href="https://typicode.github.io/husky/#/"
                    >
                      Husky 
                    </a>
                    - Husky has the ability to validate commit messages with the use commit-msg similar to pre-commit rules.
                  </li>
                  <li>
                    <a
                      href="https://commitlint.js.org/#/"
                    >
                      Commitlint 
                    </a>
                    - Linter tool for commit messages
                  </li>
                </ul>
                <h3>
                  4. Helper Modules and Libraries
                </h3>
                <ul>
                  <li>
                    <a
                      href="https://www.npmjs.com/package/env-cmd"
                    >
                      env-cmd
                    </a>
                    - A node program for executing commands using an environment from an env file.
                  </li>
                  <li>
                    <a
                      href="https://momentjs.com/docs/"
                    >
                      Moment.js
                    </a>
                    - Moment JS allows displaying of date as per localization and in human readable format.
                  </li>
                  <li>
                    <a
                      href="https://www.npmjs.com/package/react-csv"
                    >
                      react-csv
                    </a>
                    - Build CSV files on the fly basing on Array/literal object of data .
                     
                  </li>
                </ul>
                <h3>
                  5. Deployment :: For demonstraction assuming GCP for deployment
                </h3>
                <h3>
                  6. Deployment CI Followlines
                </h3>
                <ul>
                  <li>
                    Node Installation
                  </li>
                  <li>
                    Dependency Installation:: 
                    <code>
                      npm install
                    </code>
                  </li>
                  <li>
                    Linting Check :: 
                    <code>
                      npm run lint:fix
                    </code>
                  </li>
                  <li>
                    Coverage Check:: 
                    <code>
                      npm run test:production
                    </code>
                  </li>
                  <li>
                    Compiled Build Creation ::
                    <code>
                      npm run build:production
                    </code>
                  </li>
                </ul>
              </div>
            </section>
            <div
              class="d-none"
            >
              <p>
                Checking env variables value
              </p>
              <div
                class="mb-5"
              />
              <div
                class="mb-5"
              >
                development
              </div>
              <p>
                success
                 ....
                 is here to check saga & reducer connecttion
              </p>
            </div>
          </section>
          <section
            class="footer sc-bczRLJ eRSPWO"
          >
            <nav
              style="max-width: 1440px; margin: auto;"
            >
              <a
                class="footer-logo"
                href="/"
              >
                <img
                  alt="logo-img"
                  src="test-file-stub"
                  width="150"
                />
              </a>
            </nav>
            <div
              class="disclaimer-text text-center"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ips um has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
            </div>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="max-width: 100%; margin: auto;"
    >
      <div
        class="sc-hKMtZM lbYHHX dummy-template"
      >
        <header
          class="header sc-gsnTZi jOpTsK"
        >
          <nav
            class="navbar navbar-expand d-flex align-items-center justify-content-between"
          >
            <a
              class="navbar-brand "
              href="/"
            >
              <img
                alt="logo-img"
                src="test-file-stub"
                width="150"
              />
            </a>
            <div
              class="collapse navbar-collapse"
            >
              <ul
                class="navbar-nav ml-auto "
              >
                <a
                  class="nav-link"
                  href="/"
                  style="text-decoration: underline;"
                  target="_self"
                >
                  Tech Stacks
                </a>
                <a
                  class="nav-link"
                  href="/component-library"
                  style="text-decoration: none;"
                  target="_self"
                >
                  Component Library
                </a>
                <a
                  class="nav-link"
                  href="/project-structure"
                  style="text-decoration: none;"
                  target="_self"
                >
                  Project Structure
                </a>
              </ul>
            </div>
          </nav>
        </header>
        <section
          class="wrapper-std"
        >
          <section
            class="techStack sc-dkzDqf ivVfSh"
          >
            <div
              class="techStack-list"
            >
              <h2>
                Do find the tech-stack list which we are using to build this architecure
              </h2>
              <h3>
                1. Interface Design Structure
              </h3>
              <p>
                <a
                  href="https://atomicdesign.bradfrost.com/chapter-2/"
                >
                  Atomic Design Methodology
                </a>
                - Atoms, molecules, organisms, templates, and pages
              </p>
              <p>
                Atomic design allows us to see our UIs broken down to their atomic elements, and also allows us to simultaneously step through how those elements join together to form our final UIs and gives thoughtful, deliberate interface design systems.
                 
              </p>
              <h3>
                2. Framwork Dependency
              </h3>
              <ul>
                <li>
                  <a
                    href="https://reactjs.org/"
                  >
                    React
                  </a>
                   - Lightweight FE library that deals with the view layer.
                </li>
                <li>
                  <a
                    href="https://redux.js.org/"
                  >
                    Redux
                  </a>
                   - For State Management
                </li>
                <li>
                  <a
                    href="https://redux-saga.js.org/"
                  >
                    Redux Saga
                  </a>
                   - easier to manage, more efficient to execute, easy to test, and better at handling failures.
                </li>
                <li>
                  <a
                    href="https://reactrouter.com/"
                  >
                    React Router
                  </a>
                   - React Router is an API for React applications which provides dynamic, client-side routing, allows us to build a single-page web application with navigation without the page refreshing as the user navigates.
                </li>
                <li>
                  <a
                    href="https://styled-components.com/docs"
                  >
                    Styled components
                  </a>
                   - provides unique class names for your styles, thus eliminating the problems with class names duplications, misspellings, and overlaps.
                </li>
                <li>
                  <a
                    href="https://styled-components.com/docs/advanced"
                  >
                    ThemeProvider
                  </a>
                  - a wrapper component available in styled-components, we can quickly set up multiple custom themes in React and switch between them with ease.
                </li>
                <li>
                  <a
                    href="https://mui.com/material-ui/getting-started/overview/"
                  >
                    Material Ui
                  </a>
                  - CSS framework that provides React components
                </li>
                <li>
                  <a
                    href="https://react-hook-form.com/"
                  >
                    React Hook Form
                  </a>
                  - To build form components. Why React Hook Forms is because it isolates the component and avoids the other components from re-rending and thus helps improving performance also gives better validation support.
                </li>
              </ul>
              <h3>
                3. Quality Gates
              </h3>
              <ul>
                <li>
                  <a
                    href="https://testing-library.com/docs/react-testing-library/setup/"
                  >
                    React Testing Liberary with jest
                  </a>
                  - React Testing Library is a testing utility tool that's built to test the actual DOM tree rendered by React on the browser. The goal of the library is to help you write tests that resemble how a user would use your application.
                </li>
                <li>
                  <a
                    href="https://eslint.org/"
                  >
                    Eslint 
                  </a>
                  :: .eslint.config.js - ESLint is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code, with the goal of making code more consistent and avoiding bugs.
                </li>
                <li>
                  <a
                    href="https://prettier.io/docs/en/"
                  >
                    Prettier
                  </a>
                  - Styling code conforms to a consistent style.
                </li>
                <li>
                  <a
                    href="https://typicode.github.io/husky/#/"
                  >
                    Husky 
                  </a>
                  - Husky has the ability to validate commit messages with the use commit-msg similar to pre-commit rules.
                </li>
                <li>
                  <a
                    href="https://commitlint.js.org/#/"
                  >
                    Commitlint 
                  </a>
                  - Linter tool for commit messages
                </li>
              </ul>
              <h3>
                4. Helper Modules and Libraries
              </h3>
              <ul>
                <li>
                  <a
                    href="https://www.npmjs.com/package/env-cmd"
                  >
                    env-cmd
                  </a>
                  - A node program for executing commands using an environment from an env file.
                </li>
                <li>
                  <a
                    href="https://momentjs.com/docs/"
                  >
                    Moment.js
                  </a>
                  - Moment JS allows displaying of date as per localization and in human readable format.
                </li>
                <li>
                  <a
                    href="https://www.npmjs.com/package/react-csv"
                  >
                    react-csv
                  </a>
                  - Build CSV files on the fly basing on Array/literal object of data .
                   
                </li>
              </ul>
              <h3>
                5. Deployment :: For demonstraction assuming GCP for deployment
              </h3>
              <h3>
                6. Deployment CI Followlines
              </h3>
              <ul>
                <li>
                  Node Installation
                </li>
                <li>
                  Dependency Installation:: 
                  <code>
                    npm install
                  </code>
                </li>
                <li>
                  Linting Check :: 
                  <code>
                    npm run lint:fix
                  </code>
                </li>
                <li>
                  Coverage Check:: 
                  <code>
                    npm run test:production
                  </code>
                </li>
                <li>
                  Compiled Build Creation ::
                  <code>
                    npm run build:production
                  </code>
                </li>
              </ul>
            </div>
          </section>
          <div
            class="d-none"
          >
            <p>
              Checking env variables value
            </p>
            <div
              class="mb-5"
            />
            <div
              class="mb-5"
            >
              development
            </div>
            <p>
              success
               ....
               is here to check saga & reducer connecttion
            </p>
          </div>
        </section>
        <section
          class="footer sc-bczRLJ eRSPWO"
        >
          <nav
            style="max-width: 1440px; margin: auto;"
          >
            <a
              class="footer-logo"
              href="/"
            >
              <img
                alt="logo-img"
                src="test-file-stub"
                width="150"
              />
            </a>
          </nav>
          <div
            class="disclaimer-text text-center"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ips um has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
          </div>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
